includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -643,12 +684,37 @@ private function get_recurring_item_data_for_subscription( WC_Subscription $subs
 				continue;
 			}
 
-			$data[] = [
+			$item_data = [
 				'metadata'  => [ 'wc_item_id' => $item->get_id() ],
 				'price'     => $this->product_service->get_wcpay_price_id( $product ),
 				'quantity'  => $item->get_quantity(),
 				'tax_rates' => $this->get_tax_rates_for_item( $item, $subscription ),
 			];
+
+			$product_price         = (float) $product->get_price();
+			$product_interval      = (int) WC_Subscriptions_Product::get_interval( $product );
+			$product_period        = WC_Subscriptions_Product::get_period( $product );
+			$item_total            = floatval( $item->get_total() ) / $item->get_quantity();
```

Comment:
Thanks for taking another look and for the detailed explanation!

>Curious to hear your thoughts. Should we ignore that issue for now, continue with this PR and then revisit this issue in a root and branch overhaul for taxes?

I think I'd rather wait until we have a proper solution in place. Shipping in its current state with a known bug could come back to bite us in the form of angry customer support requests.

If you agree, it seems like we'd need to create a new issue to disable the "Add payment method" button for subscriptions with manual renewals. And we'd need to ship that in the MVP in place of this issue.

Is this correct?

Reply:
Oh sorry, I wasn't suggesting we ship with a critical bug. There's an open issue (#3104) for dealing with scenarios where a customer purchases a subscription outside the stores' tax location and the store has entered prices inclusive of tax. eg the store is located in California but the customer is located in Texas where a different tax applies. 

To fix that issue we will need to account for the amount of (California) tax that has been subtracted from the product price before the taxes which apply to the customer (Texas) are added. We will need to fix that issue before we ship any way given its' a critical bug too.

I was suggesting we wrap this tax issue up with that fix because this PR may be somewhat tangled up too. But I've been running some numbers and I think I might know how we can continue with this PR.

I think to fix the issue I mentioned above, you'll need to add the total tax figure (48c) onto the line item subtotal ($9.52),  and then see that matches the product price ($10).

That will work for subscriptions inside the store's tax location (eg California) but might not work for customers outside (eg Texas). However, that issue is right in #310's4 wheelhouse.

I'll run some more numbers and scenarios tomorrow to confirm that.

Reply:
> I'll run some more numbers and scenarios tomorrow to confirm that.

I couldn't resist waiting until tomorrow. 🤦‍♂️ 

This is the change I made locally and I believe this works:

```diff
@@ -694,7 +697,8 @@ class WC_Payments_Subscription_Service {
                        $product_price         = (float) $product->get_price();
                        $product_interval      = (int) WC_Subscriptions_Product::get_interval( $product );
                        $product_period        = WC_Subscriptions_Product::get_period( $product );
-                       $item_total            = floatval( $item->get_total() ) / $item->get_quantity();
+                       $inclusive_taxes       = $subscription->get_prices_include_tax() ? array_sum( $item->get_taxes()['total'] ) : 0;
+                       $item_total            = floatval( $item->get_total() + $inclusive_taxes ) / $item->get_quantity();
                        $subscription_interval = (int) $subscription->get_billing_interval();
                        $subscription_period   = $subscription->get_billing_period();
```

To replicate: 

1. Enable taxes.
2. Make sure prices are inclusive of tax. (https://d.pr/i/l7paxH)
2. Create 2 tax rates https://d.pr/i/CLBPb9. 
     - A 5% rate that applies to your stores base location (mine [is AU](https://d.pr/i/l7paxH))
     - A 10% rate that applies to some other country (I used the US).
3. Create a $10 subscription product. 
4. Purchase it through the checkout and use an Au address to the 5% inclusive tax applies.
5. On this branch, prior to that diff above it would result in the incorrect total in the Stripe dashboard .  
6. With that change it is correct (https://d.pr/i/OwqHSI) and uses the price ID, not `price_data` (https://d.pr/i/TMJqhe).

Now purchase the same subscription but this time change your address to the US so the 10% tax applies.
The total in WC for 5 x $10 product with 5% incl and 10% tax excl is $52.38 (https://d.pr/i/0XGfO8)

1. Without those changes above the totals are incorrect in Stripe (https://d.pr/i/q2pnDJ) 
2. With those changes the totals are correct (https://d.pr/i/8DThsk)

I note that I am just testing the checkout customer flow, and not the add payment method flow. Both use the same methods so should work out, however I'll run through some manually created orders and subscriptions tomorrow.